tweet_analysis/main.py

```diff
@@ -0,0 +1,104 @@
+from nltk.corpus import twitter_samples
+from nltk.tag import pos_tag
+from nltk.stem.wordnet import WordNetLemmatizer
+from nltk.corpus import stopwords
+from nltk import FreqDist
+from nltk import classify
+from nltk import NaiveBayesClassifier
+from nltk.tokenize import word_tokenize
+import re
+import string
+import random
+
+
+# Performing normalization with lemmatization and removes noise
+def remove_noise(tweet_tokens, stop_words=()):
+    cleaned_tokens = []
+    for token, tag in pos_tag(tweet_tokens):
+        token = re.sub('http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+#]|[!*\(\),]|'
+                       '(?:%[0-9a-fA-F][0-9a-fA-F]))+', '', token)
+        token = re.sub("(@[A-Za-z0-9_]+)", "", token)
+
+        if tag.startswith('NN'):
+            pos = 'n'
+        elif tag.startswith('VB'):
+            pos = 'v'
+        else:
+            pos = 'a'
+
+        lemmatizer = WordNetLemmatizer()
+        token = lemmatizer.lemmatize(token, pos)
+        if len(token) > 0 and token not in string.punctuation and token.lower() not in stop_words:
+            cleaned_tokens.append(token.lower())
+    return cleaned_tokens
+
+
+def get_all_words(cleaned_tokens_list):  # Calculating word density
+    for tokens in cleaned_tokens_list:
+        for token in tokens:
+            yield token
+
+
+# Creating a dictionary of cleaned tokens
+def get_tweets_for_model(cleaned_tokens_list):
+    for tweet_tokens in cleaned_tokens_list:
+        yield dict([token, True] for token in tweet_tokens)
+
+
+if __name__ == "__main__":
```

Comment:
I presume that you're using this as the main script for testing. However, since we're using this as a function, can make this into a function with params?

This would also make it easier for me to integrate Twint into this.

Reply:
Done!